### PR TITLE
Fix path problem on Windows - Fixes gh-2155

### LINKED ISSUE #2155: Some tests in JGitEnvironmentRepositoryTests not working on Windows
I got some test failures when building this project on Windows which works fine on Linux.

In `JGitEnvironmentRepositoryTests` I got:
```
org.opentest4j.AssertionFailedError: 
expected: "file:././target/repos/another-config-repo/sub/application.yml"
 but was: "file:././target/repos/another-config-repo/file:D:\test\spring-cloud-config\spring-cloud-config-server\target\repos\another-config-repo\sub\application.yml"
   at org.springframework.cloud.config.server.environment.JGitEnvironmentRepositoryTests.placeholderInSearchPath(JGitEnvironmentRepositoryTests.java:155)
```
```
org.opentest4j.AssertionFailedError: 
expected: "file:././target/repos/config-repo/bar.properties"
 but was: "file:././target/repos/config-repo/file:D:\test\spring-cloud-config\spring-cloud-config-server\target\repos\config-repo\bar.properties"
   at org.springframework.cloud.config.server.environment.JGitEnvironmentRepositoryTests.branch(JGitEnvironmentRepositoryTests.java:183)
```
... and other similar 6 errors. It seems they have the same path problem.

### PR DESCRIPTION
**Pull request by woshikid**
Close #2155 
fixes 8 test failures in `JGitEnvironmentRepositoryTests` on Windows

### CODE REVIEW

File: spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepository.java [hunk 1]
```diff
@@ -264,6 +264,8 @@ protected Environment clean(Environment env,
 					locations = new String[] { matcher.group(2) };
 				}
 			}
+			name = name.replace("\\", "/"); // change windows path '\' into '/'
```
Comment (ryanjbaxter) [inline spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepository.java:267]: Doesn't this effect all OSes?

Comment (woshikid) [reply spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/NativeEnvironmentRepository.java:267]: Only Windows uses `\` as path separator, so `name = name.replace("\\", "/")` only effect Windows.

**Comment by woshikid** [comment]
I was surprised, the test fails means something not working well on Windows. And the failure is not caused by the test code itself.

**Comment by ryanjbaxter** [comment]
No one on the team has a Windows machine and we run all our builds on Linux, so its certainly plausible that there are some issues on Windows.

**Pull request merged**